test/test_sync.cpp

```diff
@@ -6942,6 +6942,338 @@ TEST(Sync_SetAndGetEmptyReciprocalChangeset)
+    changeset.push_back(instr);
+    Update instr2;
+    instr2.table = changeset.intern_string("table");
+    instr2.object = instr::PrimaryKey{42};
+    instr2.field = changeset.intern_string("dict");
+    auto key = changeset.intern_string("key2");
+    instr2.path.push_back(key);
+    instr2.value = Payload{int64_t(0)};
+    changeset.push_back(instr2);
+    Clear instr3;
+    instr3.table = changeset.intern_string("table");
+    instr3.object = instr::PrimaryKey{42};
+    instr3.field = changeset.intern_string("dict");
+    instr3.collection_type = instr::CollectionType::Dictionary;
+    changeset.push_back(instr3);
+    changeset.version = 1;
+    // Make it so the merging window contains the last two local updates.
+    changeset.last_integrated_remote_version = latest_local_version - 2;
+    changeset.origin_timestamp = timestamp - 2;
+    changeset.origin_file_ident = 2;
+    server_changesets.push_back(changeset);
+
+    // Create changeset that inserts the same key as the local (discarded) update.
+    Changeset changeset2;
+    Update instr4;
+    instr4.table = changeset2.intern_string("table");
+    instr4.object = instr::PrimaryKey{42};
+    instr4.field = changeset2.intern_string("dict");
+    key = changeset2.intern_string("key");
+    instr4.path.push_back(key);
+    instr4.value = Payload{int64_t(-6)};
+    changeset2.push_back(instr4);
+    changeset2.version = 2;
+    // Make it so the merging window contains the local dictionary update.
+    changeset2.last_integrated_remote_version = latest_local_version - 1;
+    changeset2.origin_timestamp = timestamp - 1;
+    changeset2.origin_file_ident = 2;
+    server_changesets.push_back(changeset2);
+
+    std::vector<RemoteChangeset> server_changesets_encoded;
+    for (const auto& changeset : server_changesets) {
+        ChangesetEncoder::Buffer encoded;
+        encode_changeset(changeset, encoded);
+        server_changesets_encoded.clear();
+        server_changesets_encoded.emplace_back(changeset.version, changeset.last_integrated_remote_version,
+                                               BinaryData(encoded.data(), encoded.size()), changeset.origin_timestamp,
+                                               changeset.origin_file_ident);
+
+        SyncProgress progress = {};
+        progress.download.server_version = server_changesets.back().version;
+        progress.download.last_integrated_client_version = server_changesets.back().last_integrated_remote_version;
+        progress.latest_server_version.version = server_changesets.back().version;
+        progress.latest_server_version.salt = 0x7876543217654321;
+
+        uint_fast64_t downloadable_bytes = 0;
+        VersionInfo version_info;
+        auto transact = db->start_read();
+        history.integrate_server_changesets(progress, downloadable_bytes, server_changesets_encoded, version_info,
+                                            DownloadBatchState::SteadyState, *test_context.logger, transact);
+    }
+
```

Comment:
same q here, do we need to check that the reciprocal changesets are what we expect?